examples/src/Server/MonadIO/Servant.hs

```diff
@@ -0,0 +1,48 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Server.MonadIO.Servant where
+
+import Control.Concurrent.STM
+import Control.Monad.Except (ExceptT, MonadError)
+import Control.Monad.Reader (MonadReader, ReaderT)
+import Control.Monad.Trans (liftIO)
+import Control.Monad.Trans.Except (runExceptT)
+import Control.Monad.Trans.Reader (runReaderT)
+import Data.Aeson
+import Data.Morpheus (interpreter)
+import Data.Morpheus.Types
+import Data.Text (Text)
+import Network.Wai.Handler.Warp
+import Servant
+import Server.MonadIO.API hiding (Value)
+
+type GraphqlAPI = Header "Authorization" Text :> ReqBody '[JSON] GQLRequest :> Post '[JSON] GQLResponse
+
+apiServant :: GQLRequest -> Web GQLResponse
+apiServant = interpreter rootResolver
+
+graphqlAPI :: Proxy GraphqlAPI
+graphqlAPI = Proxy
+
+app :: IO ()
+app = do
+  db <- newTVarIO dbInit
+  run 8080 $ serve graphqlAPI (graphql db)
+
+graphql :: TVar Database -> Maybe Text -> GQLRequest -> Handler GQLResponse
+graphql db maybeToken request = do
+  let headers = case maybeToken of
+        Just tk -> [("Authorization", tk)]
+        _ -> []
+  let env = Env db headers
+  res <- liftIO . runExceptT . flip runReaderT env . runWeb . apiServant $ request
```

Comment:
@dandoh   to use `ServerError` we need `servant-0.16` and `servant-server-0.16`, but they can't compile with old GHV version.  `<>` is not supported as default in older version of GHC , but `/src/Servant/Server/Internal/Router.hs` uses it without importing `Data.Semigroup`. 

could you please replace ServerError with something else?